MCDataProducts/inc/CrvStep.hh

```diff
@@ -0,0 +1,65 @@
+#ifndef MCDataProducts_CrvStep_hh
+#define MCDataProducts_CrvStep_hh
+//
+// Class to summarize the passage of a single particle through a single CRV counter
+// This consolidates the G4 steps and insulates the downstream CRV response simulation 
+// from details of the G4 model
+//
+#include "canvas/Persistency/Common/Ptr.h"
+
+#include "MCDataProducts/inc/StepPointMC.hh"
+#include "MCDataProducts/inc/SimParticle.hh"
+#include "DataProducts/inc/CRSScintillatorBarIndex.hh"
+#include "DataProducts/inc/XYZVec.hh"
+#include "CLHEP/Vector/ThreeVector.h"
+#include <Rtypes.h>
+
+namespace mu2e 
+{
+  class CrvStep 
+  {
+    public:
+
+      CrvStep() : _visibleEDep(0.0), _startTime(0.0), _endTime(0.0), _pathLength(0.0) {}
+      CrvStep(CRSScintillatorBarIndex barIndex, float visibleEDep, double startTime, double endTime,
+              const XYZVec &startPos, const XYZVec &endPos, const XYZVec &startMom, const XYZVec &endMom, 
+              float pathLength, art::Ptr<SimParticle> const& simParticle) :
+              _barIndex(barIndex), _visibleEDep(visibleEDep), 
+              _startTime(startTime), _endTime(endTime), _startPos(startPos), _endPos(endPos),
+              _startMom(startMom), _endMom(endMom), _pathLength(pathLength), _simParticle(simParticle) {}
+
+      CRSScintillatorBarIndex      barIndex() const    {return _barIndex;}
+      float                        visibleEDep() const {return _visibleEDep;}
+      double                       startTime() const   {return _startTime;}
+      double                       endTime() const     {return _endTime;}
+      XYZVec                       startPos() const    {return _startPos;}
+      XYZVec                       endPos() const      {return _endPos;}
+      XYZVec                       startMom() const    {return _startMom;}
+      XYZVec                       endMom() const      {return _endMom;}
+      float                        pathLength() const  {return _pathLength;}
+      art::Ptr<SimParticle> const& simParticle() const {return _simParticle;}
+      art::Ptr<SimParticle>&       simParticle()       {return _simParticle;}
+
+    private:
+      CRSScintillatorBarIndex _barIndex;
+      float                   _visibleEDep; 
+      double                  _startTime, _endTime; //must be double to allow for long-lived particles
+      XYZVec                  _startPos,  _endPos;
+      XYZVec                  _startMom,  _endMom;
```

Comment:
Can you explain how the end momentum is used?   What is the value to storing as a vector?

Reply:
Only the magnitude of the end momentum is used (to calculate an average value of beta in the photon generator module). However, it is possible that we may find a use for the vector at a later time. I don't think that two additional floats (compared to just storing the magnitude) is a problem. However, this calculation assumes that the direction of the momentum does not change during the last StepPointMC which may not be the case. That's why I made the comment that it would be nice, if the StepPointMC provided the end momentum.